### ISSUE TITLE
[NOTICE] Not Enough Crashes can cause Iris to crash when random mods are added/removed. This is nothing to do with Notify.

### ISSUE THREAD

**Issue by efffauw** [JamCoreModding/Notify#6]
Fabric 1.17.1
Removing mod fixes it

https://paste.srnyx.xyz/ewipanawok.yaml

**Comment by Jamalam360** [comment]
Unfortunately it's currently incompatible with Iris for unknown reasons, I will check it out later

**Comment by Jamalam360** [comment]
You can see https://github.com/IrisShaders/Iris/issues/930 for information on the Iris side of the bug.

**Comment by srnyx** [comment]
@efffauw can you put your log(s) into a paste please?

**Comment by Jamalam360** [comment]
I've edited it myself to put them on your hastebin. We'll need some more input from Iris before we find out the cause of this

**Comment by Jamalam360** [comment]
@srnyx @efffauw we have an answer. Both of you have `Not Enough Crashes` present in your logs, I'll quote what was said by Coderbot in the Iris issue:

> This seems to have nothing to do with Notify or similar mods, but rather this seems to be yet another incompatibility issue with Not Enough Crashes. NEC lets the game continue initialization after a fatal entrypoint crash, and if Iris wasn't initialized before the entrypoint crash, Iris crashes trying to access state that in a SANE environment should ALWAYS BE INITIALIZED by the time LevelRenderer is getting constructed, but NEC breaks that.

TL;DR, Not Enough Crashes is incompatible with Iris. The reason this can appear/disappear when random mods are added and removed is because that changes the load order:

> Just to be clear: If you are debugging an error like this, changing the mods you have loaded will randomly change the mod loading order, meaning that adding / removing unrelated mods can cause or disable the crash.

**Comment by Jamalam360** [comment]
Fixed by Iris!
